include/swift/AST/DiagnosticsSIL.def

```diff
@@ -753,50 +753,32 @@ ERROR(sil_moveonlychecker_owned_value_consumed_and_used_at_same_time, none,
 ERROR(sil_moveonlychecker_value_used_after_consume, none,
       "'%0' used after consume", (StringRef))
 ERROR(sil_moveonlychecker_guaranteed_value_consumed, none,
-      "'%0' has guaranteed ownership but was consumed", (StringRef))
+      "'%0' is borrowed and cannot be consumed", (StringRef))
+
+// FIXME: this diagnostic shouldn't ever be emitted now. rdar://109742587 (closures may still try to consume captures, e.g., borrowed parameters)
 ERROR(sil_moveonlychecker_guaranteed_value_captured_by_closure, none,
-      "'%0' has guaranteed ownership but was consumed due to being captured by a closure", (StringRef))
-ERROR(sil_moveonlychecker_let_value_consumed_in_closure, none,
-      "'%0' consumed in closure. This is illegal since if the closure is invoked more than once the binding will be uninitialized on later invocations", (StringRef))
+      "'%0' is borrowed and cannot be consumed by closure capture", (StringRef))
+
+ERROR(sil_moveonlychecker_capture_consumed_in_closure, none,
```

Comment:
I looked through the patch and I think you are using this for escaping closures. I think there are two problems here:

1. I think the information that the closure is escaping is important for the user to understand. We should tell the user that especially since we made the change to make non-escaping lets non-escaping. That provides a lot of clarity to the user.
2. This diagnostic as shown below also occurs when a value that is captured by an escaping closure is used outside of the closure itself (the property is not flow sensitive). So the diagnostic says the use is in a closure, but it clearly isn't which is misleading.

Reply:
Seems like the main thing we need to communicate is that a mutable capture can be consumed in a closure, but an immutable one cannot. That doesn't change whether the closure is passed to an `@escaping` parameter or not.